test/integration/agent/npl_test.go

```diff
@@ -0,0 +1,132 @@
+//go:build linux
+// +build linux
+
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package agent
+
+import (
+	"testing"
+	"time"
+
+	nplcontroller "antrea.io/antrea/pkg/agent/nodeportlocal/k8s"
+	portcache "antrea.io/antrea/pkg/agent/nodeportlocal/portcache"
+	rules "antrea.io/antrea/pkg/agent/nodeportlocal/rules"
+	iptables "antrea.io/antrea/pkg/agent/util/iptables"
+)
+
+func TestNPLIptablesRestore(t *testing.T) {
```

Comment:
I think I brought this up before, but does this really need to be an integration test. I think that because PodPortRules is a mocked interface, a unit test using a mock would be just as good (an IMO the fewer OS-dependent integration tests the better).